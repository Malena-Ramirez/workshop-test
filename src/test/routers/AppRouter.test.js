import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { mount, configure } from 'enzyme';
import { AppRouter } from '../../routers/AppRouter';
import { AuthContext } from '../../auth/AuthContext';

configure({ adapter: new Adapter() });

describe('Pruebas en <AppRouter/>', () => {

  test('Debe mostrar login si el usuario NO está autenticado', () => {
    const contextValue = {
      dispatch: jest.fn(),
      user: {
        logged: false
      }
    }
    const wrapper = mount(
      <AuthContext.Provider value={contextValue}>
        <AppRouter />
      </AuthContext.Provider>
    )
    expect(wrapper).toMatchSnapshot();
  })

  test('Debe mostrar el componente de marvel si el usuario está autenticado', () => {
    const contextValue = {
      dispatch: jest.fn(),
      user: {
        logged: true,
        name: 'Thiago'
      }
    }
    const wrapper = mount(
      <AuthContext.Provider value={contextValue}>
        <AppRouter />
      </AuthContext.Provider>
    )
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('.navbar').exists()).toBe(true);
  })
})
