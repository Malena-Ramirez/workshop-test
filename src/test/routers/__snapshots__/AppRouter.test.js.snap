// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AppRouter/> Debe mostrar el componente de marvel si el usuario está autenticado 1`] = `ReactWrapper {}`;

exports[`Pruebas en <AppRouter/> Debe mostrar login si el usuario NO está autenticado 1`] = `ReactWrapper {}`;
